source/tomopy/misc/corr.py

```diff
@@ -267,6 +268,114 @@ def median_filter_cuda(arr, size=3, axis=0):
     return out
 
 
+def _determine_nonfinite_kernel_idxs(x_idx, y_idx, kernel, shape_x, shape_y):
+    """
+    Determine the proper kernel bounds for a given x, y index, and kernel size.
+
+    Parameters
+    ----------
+    x_idx : int
+        The x index for a given nonfinite value.
+    y_idx :  int
+        The y index for a given nonfinite value.
+    shape_x : int
+        The max shape of the 2D projection in the x dimension.
+    shape_y : int
+        The max shape of the 2D projection in the y dimension.
+
+    Returns
+    -------
+    x_lower, x_higher, y_lower, y_higher : int
+        The integer kernel bounds to surround the nonfinite value with.
+    """
+
+    # Determining x kernel
+    x_idx_lower = x_idx - kernel
+    x_idx_higher = x_idx + kernel + 1
+
+    if x_idx_lower < 0:
+        x_idx_lower = 0
+    if x_idx_higher > shape_x:
+        x_idx_higher = shape_x
+
+    # Determining y kernel
+    y_idx_lower = y_idx - kernel
+    y_idx_higher = y_idx + kernel + 1
+
+    if y_idx_lower < 0:
+        y_idx_lower = 0
+    if y_idx_higher > shape_y:
+        y_idx_higher = shape_y
```

Comment:
These 12 lines can be reduce to 4 lines using the built-in `min(a,b)` and `max(a,b)` functions.

Reply:
I am unsure how the min(a, b) and max(a, b) functions help. Both those functions return the absolute min and max values for an array. 

If we take `x_idx=10`, `y_idx=20`, and `kernel=1` we want values of `x_idx_lower=9`, `x_idx_higher=11`, `y_idx_lower=19`, and `y_idx_higher=21`. 

Placing these in an array of `x_array=[0, x_idx_lower, x_idx_higher, max_x_shape]`and running `min(x_array)` returns `0` when we want a return of `x_idx_lower`. 

We could do something like `x_min = min(i for i in x_array if i > 0)`, but when ` x_idx_lower<=0` then the `min()`return is `x_min=x_idx_higher`.

Reply:
For each of the 4 range limits your algorithm is equivalent to something like this:
```python
 x_idx_lower = max(0, x_idx - kernel)
```

Reply:
For clarity, `max` is the function from the standard library which compares two scalars. Not `np.max` which finds the maximum along a dimension of an array. In this context, `x_idx, y_idx` are integer scalars (as documented), so using `np.max` would not be appropriate.